packages/web/src/components/ui/data-table.tsx

```diff
@@ -37,100 +37,136 @@ export function DataTable<TData, TValue>({
     searchKey,
     searchPlaceholder,
 }: DataTableProps<TData, TValue>) {
-    const [sorting, setSorting] = React.useState<SortingState>([])
-    const [columnFilters, setColumnFilters] = React.useState<ColumnFiltersState>(
-        []
-      )
+    const [sorting, setSorting] = React.useState<SortingState>([]);
+    const [columnFilters, setColumnFilters] = React.useState<ColumnFiltersState>([]);
+    const [pagination, setPagination] = React.useState({
+        pageIndex: 0,
+        pageSize: 10, // Default page size
+    });
 
     const table = useReactTable({
         data,
         columns,
-        getCoreRowModel: getCoreRowModel(),
-        getPaginationRowModel: getPaginationRowModel(),
-        onSortingChange: setSorting,
-        getSortedRowModel: getSortedRowModel(),
-        onColumnFiltersChange: setColumnFilters,
-        getFilteredRowModel: getFilteredRowModel(),
         state: {
             sorting,
             columnFilters,
+            pagination,
         },
-    })
+        onSortingChange: setSorting,
+        onColumnFiltersChange: setColumnFilters,
+        onPaginationChange: setPagination,
+        getCoreRowModel: getCoreRowModel(),
+        getPaginationRowModel: getPaginationRowModel(),
+        getSortedRowModel: getSortedRowModel(),
+        getFilteredRowModel: getFilteredRowModel(),
+    });
 
     return (
-        <div>
-            <div className="flex items-center py-4">
+        <div className="pb-12">
```

Comment:
nit: could we add a optional `className` props to `DataTableProps`, and then have `RepositoryTable` pass down this class?